src/aal/aal_loongarch.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+#include <cstddef>
+#if _LOONGARCH_SZPTR == 64
+#  define SNMALLOC_VA_BITS_64
+#elif _LOONGARCH_SZPTR == 32
+#  define SNMALLOC_VA_BITS_32
+#else 
+#  error "Unsupported pointer size"
+#endif
+namespace snmalloc
+{
+  /**
+   * ARM-specific architecture abstraction layer.
+   */
+  class AAL_loongarch
```

Comment:
Better use consistent casing as with the enum variant:

```suggestion
  class AAL_LoongArch
```